utils/fix_formatting.py

```diff
@@ -30,49 +30,75 @@ def parse_args():
         ' over the code base and pyformat over the utils directory.'
         ' Recurses over all relevant files.')
-    return p.parse_args()
+
+    p.add_argument('-c', '--c++', action='store_true',
+                   help='Format C++ files only')
+    p.add_argument('-l', '--lua', action='store_true',
+                   help='Format Lua files only')
+    p.add_argument('-p', '--python', action='store_true',
+                   help='Format Python files only')
+    p.add_argument('-d', '--dir', action='store',
+                   help='Format the given directory and its subdirectories only')
+
+    return vars(p.parse_args())
 
 
 def main():
-    parse_args()
+    args = parse_args()
+    format_cplusplus = args['c++'] or not (args['lua'] or args['python'])
+    format_lua = args['lua'] or not (args['c++'] or args['python'])
+    format_python = args['python'] or not (args['c++'] or args['lua'])
 
     if not os.path.isdir('src') or not os.path.isdir('utils'):
         print('CWD is not the root of the repository.')
         return 1
 
-    sys.stdout.write('\nFormatting C++ ')
-    for filename in find_files('./src', ['.cc', '.h']):
-        if 'third_party' in filename:
-            continue
-        sys.stdout.write('.')
-        sys.stdout.flush()
-        call(['clang-format', '-i', filename])
-        call(['git', 'add', '--renormalize', filename])
-    print(' done.')
-
-    sys.stdout.write('Fixing Lua tabs ')
-    for filename in find_files('.', ['.lua']):
-        sys.stdout.write('.')
-        sys.stdout.flush()
-        lines = read_text_file(filename).strip().split('\n')
-        new_lines = []
-        for line in lines:
-            m = LEADING_TABS.match(line)
-            if m is not None:
-                line = line[m.start():m.end()].expandtabs(
-                    SPACES_PER_TAB) + line[m.end():]
-            new_lines.append(line.rstrip() + '\n')
-        write_text_file(filename, ''.join(new_lines))
-        call(['git', 'add', '--renormalize', filename])
-    print(' done.')
-
-    sys.stdout.write('\nFormatting Python utils ')
-    for filename in find_files('./utils', ['.py']):
-        sys.stdout.write('.')
-        sys.stdout.flush()
-        call(['pyformat', '-i', filename])
-        call(['git', 'add', '--renormalize', filename])
-    print(' done.')
+    if format_cplusplus:
+        directory = args['dir']
+        if not directory:
+            directory = './src'
+        sys.stdout.write('\nFormatting C++ in directory: ' + directory + ' ')
+        for filename in find_files(directory, ['.cc', '.h']):
+            if 'third_party' in filename:
+                continue
+            sys.stdout.write('.')
+            sys.stdout.flush()
+            call(['clang-format', '-i', filename])
+            call(['git', 'add', '--renormalize', filename])
+        print(' done.')
+
+    if format_lua:
+        directory = args['dir']
+        if not directory:
+            directory = './data'
```

Comment:
We also have Lua files in `./test` that should be formatted

Reply:
Thanks for catching this one. I am using a set now. Same for Python, because we should format the codecheck rules too in the future. I'd like to review the formatting results though to ensure that they are good, so I'd rather add that directory in a follow-up branch.